### PR TITLE
Cleaning up VCO

### PR DESCRIPTION
**Pull request by turian**
@maxsolomonhenry there are a few things I need help with here, if it's a simple fix just push changes into this branch, or respond to comments if it's more discussion or there's something didactic for me.

### CODE REVIEW

File: src/ddspdrum/module.py [hunk 1]
```diff
@@ -173,31 +175,30 @@ def __init__(
...
-
-    def set_phase(self, val):
-        self.__phase = val % (2 * np.pi)
-
-    def play(self):
-        arg = self.to_arg(self.get_f0()) + self.get_phase()
-        self.set_phase(arg[-1])
+    def __call__(self, f0control: np.array, phase: float):
+        # TODO: Are there expected ranges on phase?
+        # Does it matter? Can we just assert preconditions here?
```
Comment (turian) [inline src/ddspdrum/module.py:180]: Let's start enforcing preconditions on parameter values.

Comment (turian) [reply src/ddspdrum/module.py:180]: It's important for making sure that randomly generated patches are sensible

File: src/ddspdrum/module.py [hunk 2]
```diff
@@ -173,31 +175,30 @@ def __init__(
...
+        # What is arg and can we give it a better name?
+        arg = self.to_arg(clampedf0signal) + phase
+        # Why do we do a `set_phase` here at the end?
+        # If this is really some sort of recurrent state then maybe
+        # we return it and pass it back in?
+        #self.set_phase(arg[-1])
         return np.cos(arg)
 
-    def to_arg(self, f0):
+    def to_arg(self, f0control: np.array):
```
Comment (turian) [inline src/ddspdrum/module.py:197]: @maxsolomonhenry can you please make this easier to understand? Maybe a better function name and docstring?

File: src/ddspdrum/module.py [hunk 3]
```diff
@@ -173,31 +175,30 @@ def __init__(
...
+	    # because we lose the gradients from it, but let's worry
+        # about that later.
+        clampedf0control = np.clamp(f0control, 0, nyquist)
+
+        # What is arg and can we give it a better name?
+        arg = self.to_arg(clampedf0signal) + phase
+        # Why do we do a `set_phase` here at the end?
+        # If this is really some sort of recurrent state then maybe
+        # we return it and pass it back in?
+        #self.set_phase(arg[-1])
```
Comment (turian) [inline src/ddspdrum/module.py:194]: @maxsolomonhenry this is what's really confusing me. Why do we have `self.set_phase(arg[-1])` at the end here?

Comment (turian) [reply src/ddspdrum/module.py:194]: Why does it need to save its phase?

Comment (maxsolomonhenry) [reply src/ddspdrum/module.py]: For the next output -- it has to start up at the previous phase value or else you'll get a click.

File: src/ddspdrum/module.py (hunk 2 above)
Comment (maxsolomonhenry) [reply src/ddspdrum/module.py:197]: Frequency is the derivative of phase. Phase is an integration of frequency. "argument" is what goes in the cosine, it's the phase.

File: src/ddspdrum/module.py (hunk 1 above)
Comment (turian) [reply src/ddspdrum/module.py:180]: simple asserts

**Pull request merged**